pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -35,13 +35,15 @@ type NodePhase string
 
 // Workflow and node statuses
 const (
-	NodePending   NodePhase = "Pending"
-	NodeRunning   NodePhase = "Running"
-	NodeSucceeded NodePhase = "Succeeded"
-	NodeSkipped   NodePhase = "Skipped"
-	NodeFailed    NodePhase = "Failed"
-	NodeError     NodePhase = "Error"
-	NodeOmitted   NodePhase = "Omitted"
+	NodePending   NodePhase = "Pending" // WF/Node is waiting to run
+	NodeRunning   NodePhase = "Running" // WF/Node is currently running
+	NodeSucceeded NodePhase = "Succeeded" // WF/Node has finished with no errors
+	NodeSkipped   NodePhase = "Skipped" // Node was skipped in Workflow DAG / Steps
+	NodeFailed    NodePhase = "Failed" // a WF with a failed Node will be marked with NodeFailed as well
+	// There was an error within the container. E.g. the container exits with a non 0 code
+	NodeError     NodePhase = "Error" // a WF with an errored Node will be marked with NodeError as well
+	// There was an error around a container, E.g. The pod was deleted, couldn’t be scheduled, or there were problems getting logs
```

Comment:
I don't think this is correct - @simster7

Reply:
Seems correct to me. What do you think is wrong @alexec?